### DISCUSSION TITLE
proxy_redirect question

### DISCUSSION THREAD

**Discussion by dimon-v** [apache/apisix#5757] [Q&A]
apisix version: 2.9
linux : 3.10.0-1160.42.2.el7.x86_64

back-end nginx version: 
Tengine version: Tengine/2.3.2
nginx version: nginx/1.17.3

The production environment has an authentication interface using wechat oAuth2.0. After using Apisix, the requested host will be overwritten.  Instead of using Apisix, add 'proxy_redirect off' to ngx1 proxy_pass;  'You can reproduce the scene.  

![image](https://user-images.githubusercontent.com/16206290/145370437-6acfbdcc-5d11-44e3-93b7-bec1b24100d2.png)

**Comment by dimon-v** [comment]
apisix route:
![image](https://user-images.githubusercontent.com/16206290/145374403-12019b85-4405-464e-ab15-b4366eaf14cf.png)

**Comment by juzhiyuan** [comment]
Maybe @membphis could give some tips?

**Reply by membphis** [reply]
@dimon-v you need to specify the upstream host name.

take a look at the `pass_host` of https://github.com/apache/apisix/blob/master/docs/en/latest/admin-api.md#upstream
